activerecord/lib/active_record/tasks/database_tasks.rb

```diff
@@ -142,8 +142,10 @@ def create_all
       end
 
       def for_each
-        databases = Rails.application.config.load_database_yaml
-        database_configs = ActiveRecord::DatabaseConfigurations.new(databases).configs_for(env_name: Rails.env)
+        databases = defined?(Rails) ? Rails.application.config.load_database_yaml : []
```

Comment:
I can see why you're passing the empty array here since `build_configs_will` just return the empty array.  But given that it will return just an empty array we know that the rest of this function will just no-op, perhaps it's better to just have a guard clause here: `return unless defined?(Rails)`. 

I think a better approach might be even to have something like: 

``` ruby
databases = if defined?(Rails) ? Rails.application.config.load_database_yaml  : database_configuration

return unless databases
```

That way if someone has set the `databse_configuration` accessor we use it, instead of having a empty array, and we also return early if no configuration is set.

Reply:
I wanted to followup here - when loading these tasks `database_configuration` will always be `nil` because they are loaded before the application is booted and therefore before Railties loads `database_configuration` into Active Record. In my PR I chose to just ignore the multi-db tasks altogether unless Rails is defined.